test/e2e/set/scyllacluster/helpers.go

```diff
@@ -350,7 +364,15 @@ func restartRack(ctx context.Context, client appv1client.AppsV1Interface, r scyl
 		return err
 	}
 
-	_, err = client.StatefulSets(sc.Namespace).Patch(ctx, sts.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+	_, err = appClient.StatefulSets(sc.Namespace).Patch(ctx, sts.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+	if err != nil {
+		return err
+	}
+
+	_, err = waitForStatefulSetState(ctx, appClient, sts.Namespace, sts.Name, func(sts *appsv1.StatefulSet) (bool, error) {
```

Comment:
Aren't you already waiting on the call sites at L113 and L138?

Reply:
I'm, for opposite state. The flake was that after adding annotation to the sts, which is supposed to trigger the rollout, test verified sts rollout before sts.Status.UpdateRevision was updated by sts controller. Take a look at the timestamps:
```
// annotate sts
Jun 17 10:34:21.809: STEP: Initiating a rolling restart 
// wait for sts rollout
Jun 17 10:34:21.836: STEP: Waiting for the ScyllaCluster to pick up token change 
// sts rolled out, check api auth
Jun 17 10:34:21.841: STEP: Accepting requests authorized using token from user agent config 
```
So here, I'm waiting for the rollout to start.

Reply:
isn't the opposite state immediately true, same way the actual state is guaranteed to be immediately false as admission raises `generation` on template change and the `observedGeneration` doesn't match until statefulset controller updates the status?

Btw. now that we have audit you could list all API calls on the statefulset (with timestamps and content) and confirm any theory.